pf2/predict.py

```diff
@@ -76,35 +84,44 @@ def predict_mortality(data: pd.DataFrame, meta: pd.DataFrame, proba: bool = Fals
             labels (pd.Series): classification targets
         else:
             accuracy (float): prediction accuracy
-            coefficients (pd.DataFrame): LR model coefficients; columns
-                correspond to unique models for COVID/non-COVID patients
+            models (tuple[COVID, Non-COVID]): fitted PLSR models
     """
     if not isinstance(data, pd.DataFrame):
         data = pd.DataFrame(data)
 
-    data = data.loc[meta.loc[:, "patient_category"] != "Non-Pneumonia Control", :]
-    meta = meta.loc[meta.loc[:, "patient_category"] != "Non-Pneumonia Control", :]
+    data = data.loc[
+        meta.loc[:, "patient_category"] != "Non-Pneumonia Control",
+        :
+    ]
+    meta = meta.loc[
+        meta.loc[:, "patient_category"] != "Non-Pneumonia Control",
+        :
+    ]
     labels = data.index.to_series().replace(meta.loc[:, "binary_outcome"])
-    groups = meta.loc[:, "patient_category"] == "COVID-19"
-    groups = groups.astype(int)
 
-    predictions = pd.Series(index=data.index)
-    coefficients = pd.DataFrame(index=data.columns, columns=np.unique(groups))
-
-    for group in groups.unique():
-        group_data = data.loc[groups == group, :]
-        group_labels = labels.loc[groups == group]
+    covid_data = data.loc[meta.loc[:, "patient_category"] == "COVID-19", :]
+    covid_labels = meta.loc[
+        meta.loc[:, "patient_category"] == "COVID-19",
+        "binary_outcome"
+    ]
+    nc_data = data.loc[meta.loc[:, "patient_category"] != "COVID-19", :]
+    nc_labels = meta.loc[
+        meta.loc[:, "patient_category"] != "COVID-19",
+        "binary_outcome"
+    ]
 
-        group_predictions, group_coefficients = run_lr(
-            group_data, group_labels, proba=proba
+    predictions = pd.Series(index=data.index)
+    predictions.loc[meta.loc[:, "patient_category"] == "COVID-19"], c_plsr = \
+        run_plsr(
+            covid_data, covid_labels, proba=proba, n_components=n_components
+        )
+    predictions.loc[meta.loc[:, "patient_category"] != "COVID-19"], nc_plsr = \
+        run_plsr(
+            nc_data, nc_labels, proba=proba, n_components=n_components
         )
-        predictions.loc[groups == group] = group_predictions
-        coefficients.loc[:, group] = group_coefficients
-
-    coefficients.columns = ["Non-COVID", "COVID-19"]
 
     if proba:
         return predictions, labels
     else:
         predicted = predictions.round().astype(int)
-        return accuracy_score(labels, predicted), coefficients
+        return accuracy_score(labels, predicted), (c_plsr, nc_plsr)
```

Comment:
@JacksonLChin Should there be an accuracy score for both C19 and nC19 patients?

Reply:
This is setup to report one accuracy score across the two models--it'll use the nC19 model to predict nC19 patients and the C19 model to predict C19 patients, then combine predictions into one array to evaluate prediction accuracy. There's a way to also recover prediction accuracies for each model separately through the `proba=True` option.